src/TimeSync.cpp

```diff
@@ -52,38 +52,35 @@ namespace time_sync {
                     SPDLOG_DEBUG("TimeSync initialized!");
                 }
                 auto message = readTimeSyncMessage(time_sync);
-                clock->update(message.timestamp);
+                ClockSingleton::update(message.timestamp);
                 if (_performance_logging) {
                     if(auto logger = spdlog::get("performance"); logger != nullptr ){
                         logger->info("{0},{1}", 
                             std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count(), 
-                            clock->nowInMilliseconds());
+                            ClockSingleton::time_in_ms());
                     }
                 }
             }
         }
     }
-    unsigned long TimeSync::nowInMilliseconds() const{
-        clock->wait_for_initialization();
-        return clock->nowInMilliseconds();
-    }
-
-    void TimeSync::sleep(unsigned long ms) {
-        clock->wait_for_initialization();
-        clock->sleep_for(ms);
-    }
-
-    void TimeSync::sleepUntil(unsigned long ms) {
-        clock->wait_for_initialization();
-        clock->sleep_until(ms);
-    }
+    
     void TimeSync::stop() {
         _running = false;
         if (consumer_thread.joinable()) {
             consumer_thread.join();
         }
     }
+    unsigned long nowInMilliseconds(){
```

Comment:
Do we still needs these functions in timeSync as timeSync can/already directly use the ClockSingleton to call its corresponding static funcs.

Reply:
We could remove these and ask developers to directly call clock singleton, but I think there is value in abstracting this. This way the user does not need to know at all about the underlying singleton. The user simply need to replace calls to system time  with these. Do you think this would be cleaner without these methods and why?

Reply:
I see the TineSync directly use ClockSingleton rather than the defined functions below in the same namespace, for example:  ClockSingleton::time_in_ms() is called rather than nowInMilliseconds()?